modules/proxyconnection/ProxyConnectionService.js

```diff
@@ -0,0 +1,301 @@
+     * for more details.
+     * @private
+     * @returns {ProxyConnectionPC}
+     */
+    _createPeerConnection(peerJid, options = {}) {
+        if (!peerJid) {
+            throw new Error('Cannot create ProxyConnectionPC without a peer.');
+        }
+
+        const pcOptions = {
+            onError: this._onFatalError,
+            onRemoteStream: this._onRemoteStream,
+            onSendMessage: this._onSendMessage,
+            peerJid,
+            ...options
+        };
+
+        return new ProxyConnectionPC(pcOptions);
+    }
+
+    /**
+     * Callback invoked when an error occurs that should cause
+     * {@code ProxyConnectionPC} to be closed if the peer is currently
+     * connected. Sends an error message/reply back to the peer.
+     *
+     * @param {string} peerJid - The peer jid with which the connection was
+     * attempted or started, and to which an iq with error details should be
+     * sent.
+     * @param {string} errorType - The constant indicating the type of the error
+     * that occured.
+     * @param {string} details - Optional additional data about the error.
+     * @private
+     * @returns {void}
+     */
+    _onFatalError(peerJid, errorType, details = '') {
+        logger.error(
+            'Received a proxy connection error', peerJid, errorType, details);
+
+        const iq = $iq({
+            to: peerJid,
+            type: 'set'
+        })
+            .c('jingle', {
+                xmlns: 'urn:xmpp:jingle:1',
+                action: errorType
+            })
+            .c('details')
+            .t(details)
+            .up();
+
+        this._onSendMessage(peerJid, iq);
+
+        if (this._peerConnection
+            && this._peerConnection.getPeerJid() === peerJid) {
+            this._selfCloseConnection();
+        }
+    }
+
+    /**
+     * Callback invoked when the remote peer of the {@code ProxyConnectionPC}
+     * has offered a media stream. The stream is converted into a
+     * {@code JitsiLocalTrack} for local usage if the {@code onRemoteStream}
+     * callback is defined.
+     *
+     * @param {JitsiRemoteTrack} jitsiRemoteTrack - The {@code JitsiRemoteTrack}
+     * for the peer's media stream.
+     * @private
+     * @returns {void}
+     */
+    _onRemoteStream(jitsiRemoteTrack) {
+        if (!this._options.onRemoteStream) {
+            logger.error('Remote track received withou callback to handle it.');
+
+            return;
+        }
+
+        // Grab the webrtc media stream and pipe it through the same processing
+        // that would occur for a locally obtained media stream.
+        const mediaStream = jitsiRemoteTrack.getOriginalStream();
+        const jitsiLocalTracks = RTC.newCreateLocalTracks(
+            [
+                {
+                    deviceId:
+                        `proxy:${this._peerConnection.getPeerJid()}`,
+                    mediaType: 'video',
+                    videoType: 'desktop',
```

Comment:
So this looks very specific to our case given that the video type is hardcoded as 'desktop'. Then you can probably disregard my other comment about `receiveAudio`.

Reply:
This proxy connection is definitely tailored for jitsi spot use case only. I'd be surprised, and a little afraid, if people really started using this proxy. That said, I tried making these fields more configurable.